src/nomad_simulations/schema_packages/model_system.py

```diff
@@ -1095,7 +1113,142 @@ class ModelSystem(System):
         """,
     )
 
-    model_system = SubSection(sub_section=SectionProxy('ModelSystem'), repeats=True)
+    total_charge = Quantity(
+        type=np.int32,
+        description="""
+        Total charge of the system.
+        """,
+    )
+
+    total_spin = Quantity(
+        type=np.int32,
+        description="""
+        Total spin state of the system.
+        Not to be confused with the spin multiplicity 2S + 1.
+        """,
+    )
+
+    # New subsection: list of unique atom definitions
+    atom_types = SubSection(
+        section_def=AtomDefn.m_def,
+        repeats=True,
+        description='Unique list of atoms for the simulation.',
+    )
+
+    # New subsection: list of atomic state entries
+    atom_states = SubSection(
+        section_def=AtomsState.m_def,
+        repeats=True,
+        description='List of atomic elements (and their electronic state) for the simulation.',
+    )
+
+    particle_states = SubSection(
```

Comment:
I might be fine with this, since I guess there will only ever be either atomic or non-atomic...

but why not just have 
class ParticleState():
   ...some generic stuff
class AtomsState(ParticleState):
   ...atom specific stuff
class CoarseGrainedState(ParticleState):
   ...for a CG model

particle_states = SubSection(...ParticleState.m_def)

Then if a new sub-class comes along it's easier to extend. Is there some obvious downside?

Same for the Defs().

I guess from the normalizing standpoint it's a little more difficult because you might have to check the child class type or something to execute some logic. I think I found this when working with the trial implementation of ParticleCell, but I can't exactly remember.